node/flatpak-node-generator.py

```diff
@@ -1355,10 +1447,16 @@ class ProviderFactory:
     def create_lockfile_provider(self) -> LockfileProvider:
         raise NotImplementedError()
 
+    def create_rcfile_provider(self):
+        raise NotImplementedError()
+
     def create_module_provider(self, gen: ManifestGenerator,
                                special: SpecialSourceProvider) -> ModuleProvider:
         raise NotImplementedError()
 
+    def create_headers_provider(self, gen: ManifestGenerator) -> NodeHeadersProvider:
```

Comment:
ProviderFactory is mostly for stuff that's package-manager-dependent, I don't think the node headers really fit here? (though for the record rcfile provider probably does) If they do then this impl should probably be moved to NpmProviderFactory/YarnProviderFactory and then this is just a NotImplemented superclass

Reply:
Sorry, not sure if I understood. Should I remove `create_headers_provider()` altogether and use `NodeHeadersProvider` directly?